### ISSUE TITLE
Explorer / Opened Editors Sorting

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#27286]
Numerous issues have been filed on this topic, merging them into one:
* Add option to sort files in Explorer side-bar by date modified #11823
* Feature Request: api to custom files sort order at the sidebar #12345
* Feature request - Add sorting to Open Editors list #12453
* Can't sort files by date modified in open folder #13029
* Allow to sort list of opened editors #15317
* Please add options for sorting and choosing which file type to view in the Explorer panel #17828
* Add a setting for case sensitive sorting in explorer #18622
* Feature request: sort `_` after `.` in the file list #23007
* Choose a sort order of directories and files according to the settings pattern #23231
* Feature to assign a particular file name to appear first in a directory listing #25724
* Allow to customize explorer file sort order #26921

**Comment by dineshrajan** [comment]
Want to add one more;
If explorer have 100's of files, scrolling down to find the right one is painful. Will be great if we can type in the file or first few chars and explorer will scroll down to the right file

**Comment by csheets** [comment]
agree with the "file filtering" feature - with and without sorting

**Comment by ConnorsFan** [comment]
I would like to have an option to sort the opened editors by alphabetical order. To be honest, I don't know why that is not the default. The current sort order doesn't make sense to me (and never helps me to find what I want).

**Comment by Zyphrax** [comment]
Working with TypeScript, it would be great to have the barrels (index.ts) sorted at the top.
Would that be possible?

**Comment by slowaways** [comment]
Everything to complete an IDE is perfect for me! Organization is all in one project!

**Comment by Macromeda** [comment]
can we have an option to write our own order in json array inside .vscode ? it would make it super easy, the order will follow the order in the array

**Comment by slowaways** [comment]
@Macromeda Check my issue #23231

**Comment by Macromeda** [comment]
@slowaways my idea is to make a json that holds the actual names of the files in the project folder, not ordering by characters.

**Comment by charlesr1971** [comment]
We really need this feature. Alphabetical sorting in the Open Editors panel. Thanks...

**Comment by will-stone** [comment]
Ordering by path would be my ideal. When working with huge repos, sorting by alphabetical filename wouldn't cut it as I like to keep my editors organised by where they appear in the codebase.

e.g. this...
```
Button.js src/components
Card.js src/components
Input.js src/components
Navigation.js src/components
Auth.js src/redux
Peopls.js src/redux
User.js src/redux
map.js src/utils
parse.js src/utils
sort.js src/utils
```
would be much more useful (to me) than this...
```
Auth.js src/redux
Button.js src/components
Card.js src/components
Input.js src/components
map.js src/utils
Navigation.js src/components
parse.js src/utils
Peopls.js src/redux
sort.js src/utils
User.js src/redux
```

This is especially when you are working on goliath monorepos.

**Comment by lrkwz** [comment]
I cannot understand why the ordering is so strange: why the 2016*xml files are listed **after** 2018*xml ?

![image](https://user-images.githubusercontent.com/1025560/44461037-a3f0fd00-a60f-11e8-8490-f8ac47e0544e.png)

**Comment by charlesr1971** [comment]
Maybe the default ordering is TextNoCase ASC? A-Z. It is parsing the file name as text rather than a numeric?

**Comment by lrkwz** [comment]
Looks like is parsing as numeric instead; has is the output of `ls | sort -n`

**Comment by coling** [comment]
I don't know if this has been suggested before (or even if it's possible with an extension) but I always prefer to keep the open editors list managed as a mini-tree. i.e. identical to the main folder/files views, but only for open files. I appreciate you can see the folder paths after the file names, but the code I work on has so many similar/identically named files (where only the path differs) that this "extra" data is actually the most important part to me.

**Comment by weinand** [comment]
I've filed this issue against the strange sort order: https://github.com/Microsoft/vscode/issues/67154

**Comment by GlenPerkins** [comment]
**Use case:** I have writing projects (ex: user manual, journal article, book) with sections, sub-*-sections, files. I frequently reorganize the sequence as I work, drag, insert, delete, regroup. I want my work in the explorer to be in the desired sequence without having to rename anything.

Probably no one "best" way to do this, so create an API through which the item-by-item sequence of items in the explorer can be specified, which will allow extensions to provide various options. It should probably also enable a little sort menu so that the sort can be switched with just a click: default <> global:myjsprojects <> global:mytaxprojects <> recents <> local:custom1 <> local:custom1:todo 

I would *probably* like direct drag-and-drop reordering (that remembered my custom order when switching between custom and other sort orders), but that might risk hard-coding something someone doesn't like. Maybe I wouldn't like it. So, I'd probably be just as happy with a `FooSort` extension that had me sort the items in its own `fooSortOrder.config` files that could be global and/or local. I could drag and drop items in those almost as easily (as long as it wasn't JSON!), and a file like that could have more sort options. Different extensions could experiment with different sort order specification languages in their own config files (these files at top & bottom, others alpha in middle; or this folder numeric, this folder custom, default folder alpha; or whatever).

**Comment by SimonDever** [comment]
I've just coded a barrel-first sort order option for the explorer view that, while not providing the customization levels asked for in this issue, addresses one desire I'm sure I'm not the only one to have: sorting barrel files (index.ts, index.js, etc.) below folders but above all other files. Additionally, I couldn't find the API and don't believe it exists to perform this in an extension. I'd be happy  to change this into an extension if someone knows how to either extend or patch the explorerViewer. Reproducing File Explorer with Tree View in an extension for this one additional function I feel is overkill so this is why the pull request.

**Comment by GlenPerkins** [comment]
I agree that when I write TypeScript, I want index.ts on top, and for C, I want <somename>.c (the entry point) on top. When I write the user guide, I want my outline on top, when writing a book, I want an outline on top of each section. Whatever the definition of barrel file may be, it will either be hard-coded to the wrong choice for what many people want on top in many cases, or it will be configurable, in which case you may as well put your configuration in a config file that offers more options than just top file but doesn't require them.

I'm finding now that even the annoying workaround of renaming files to force a display sequence via alphabetical order isn't predictable with underscores, casing, extensions, different OSes, etc., making me wish I could configure what "alphabetical" means to me. A display order API for extensions would enable extensions to configure that, too.

**Comment by leidegre** [comment]
A trivial sort order which isn't without merit is `none` (or `preserve`). It should result in file system dependent sort order. For NTFS this is largely a lexicographic sort order but I know this isn't the case with all file systems. But it requires that the order of files in array from `readdir` is never changed by any code that touches the array.

**Comment by gpinkham** [comment]
But the open editors list isn't tied to readdir.      as an example I just opened a project. no files are open yet.    hit cmd-n to open a new untitled/unsaved editor..       did this again.     clicked on untitled-1.    cmd-n again.    clicked on untitled-1 again and cmd-n a 4th time.    the order in open editors matches the order of the tabs: untitled-1, untitled-4, untitled-3, untitled-2..

**Comment by leidegre** [comment]
> But the open editors list isn't tied to readdir. as an example I just opened a project. no files are open yet. hit cmd-n to open a new untitled/unsaved editor.. did this again. clicked on untitled-1. cmd-n again. clicked on untitled-1 again and cmd-n a 4th time. the order in open editors matches the order of the tabs: untitled-1, untitled-4, untitled-3, untitled-2..

Wrote it in reply to file explorer sorting. As I understood this is the thread after consolidation. I could be wrong.

**Comment by Bakerbakura** [comment]
I've also been trying to order the files in my explorer window, and having to deal with the fact that the sorting is not actually alphanumeric/lexicographic!
For instance, the file `51_lattices_metric.tex` appears *after* the file `6_modular_forms.tex`, whereas if I change it to `5_1_lattices_metric.tex` it moves *before* the file `5_lattices.tex`; what I'd really like is for the `lattices_metric.tex` file to be *between* `5_lattices.tex` and `6_modular_forms.tex`.

**Comment by SimplyInk** [comment]
Managed to reproduce OPEN EDITORS not sorting when dragging files into VS Code.

https://github.com/microsoft/vscode/issues/80681#issuecomment-539862813

**Comment by wensonsmith** [comment]
two years later , this still pending ?

**Comment by SimonDever** [comment]
@isidorn can you escalate this to a higher level? Who is the product owner that decides what the feature details should be? The code isn't that hard to write and I have the time, just tell me why my PR isn't enough and I'll enhance it.

**Comment by ojde** [comment]
+1

**Comment by reynaldots** [comment]
Sort tabs opened: https://marketplace.visualstudio.com/items?itemName=omagerio.tabsort

**Comment by woytowg** [comment]
THANK YOU!!!

Reynaldo wrote on 1/21/20 6:55 PM:

**Comment by nengxu** [comment]
I'm currently using this extension for sorting opened files:
https://marketplace.visualstudio.com/items?itemName=Divni.open-files

**Comment by nengxu** [comment]
Sorry, I found that 'tabsort' extension is not working. I have to switch back to "Open-files".

**Comment by woytowg** [comment]
Works fine for me, enable the extension close all editors, then start 
opening again, and they are sorted

nengxu wrote on 1/23/20 4:30 PM:

**Comment by jeff-hykin** [comment]
Unlike many of the other requests, I'd like to propose a small change and have made commit showing the proposed changes on a VS Code fork. https://github.com/jeff-hykin/vscode/commit/70fb0f8eec170db4ddc274be0089d20dcaa9250e 

### I've been redirected to this issue/thread, so here is my original request

## Why is this a problem (pseudo-bug)
Sorting files by both `filesFirst` and `modified` (timeModified) is certainly possible, but VS Code treats them as mutually exclusive. It seems very strange to me to include a `mixed` option,  and `modified` and option, but not a `mixed--modified` option.

For example, here's what the current VS Code options could more aptly be named:
`filesFirst--alphabetically`
`mixed--alphabetically`
`foldersFirst--alphabetically`
`foldersFirst--timeModified`
`foldersFirst--type`
But for obvious reasons switching to a naming scheme like that would quite a bit less than ideal.

## A Backwards Compatible, Future-Proofed, ~20-line API Proposal

I'd like to propose that `explorer.sortOrder` optionally accept an array of sort methods.
`explorer.sortOrder: SortMethod | SortMethod[]`
```
enum SortMethod {
    filesFirst   = "filesFirst",
    mixed        = "mixed",
    default      = "default",
    modified     = "modified",
    type         = "type",

    // additional options (see below for justification)
    foldersFirst = "foldersFirst",
    alphabetical = "alphabetical", 
}
```

## Behavior and Examples
```jsonc
{
    "explorer.sortOrder": [ "type" ]   // equivalent to "mixed--type" 
}
```
```jsonc
{
    "explorer.sortOrder": [ "type", "alphabetical"  ]
     // files are mixed (by default)
     // files then primarily sorted by type
     // files with the same type are then sorted alphabetically
}
```
```jsonc
{
    "explorer.sortOrder": [  "filesFirst", "type", "modified"  ]
     // files primarily sorted with files at top
     // files and folders are sorted by extension name
     // files with the same extension name are then sorted by timeModified
}
```
## Example Implementation

https://github.com/jeff-hykin/vscode/commit/70fb0f8eec170db4ddc274be0089d20dcaa9250e 

## Why this is ideal
- The API change is about as minimal as an API change can possibly be
- The naming will still make sense (arguably will make more sense after the change)
- The additional codebase complexity is minimal and modular
- Makes space for future functionality such as `timeCreated` or descending/ascending sorts
- It allows the user to make the choice for a desirable customization at the cost of some performance (many of us have plenty of performance to spare)

**Comment by angularorbit** [comment]
I want the "Feature Request: api to custom files sort order at the sidebar".

**Comment by kriswuollett** [comment]
I have Explorer Sort Order set to `default` in version 1.45.1 on macOS. I've got a folder of files that have a hex code prefix of fixed length from using Alembic SQL migrations. It is quite frustrating that the sort order isn't the same as using `ls` in a terminal which lets me see the migration files in alphabetic (increasing hex) order. I can't quickly click on the file to open one without searching files by typing in the hex code. This issue is similar to that in https://github.com/microsoft/vscode/issues/27286#issuecomment-415001567 above.

**Comment by planetrocky** [comment]
> I would like to have an option to sort the opened editors list by alphabetical order. To be honest, I don't know why that is not the default. The current sort order doesn't make sense to me (and never helps me to find what I want).

Yes, yes, yes!  Open Editors tab is almost useless to me; I have hundreds of source files open, really need a sort option, been waiting on this feature for *year* now!

**Comment by marcusviar** [comment]
Sort tabs alphabetically and group by file extension:

https://marketplace.visualstudio.com/items?itemName=marcusviar.sort-tabs

Please read the know issues section of the readme before use.

**Comment by seiyria** [comment]
It took me almost an hour to find this issue! I would like to echo the others in this thread: I want to move my `index` files below folders, but above other files. Have any API changes been made where an extension could do this? I would like to create an extension if so. Alternatively, maybe a new `workspace.sortOrder` that mentions something about `indexFirst`? I'm not sure what would be best, but I'd love to help if possible.

**Comment by SimonDever** [comment]
@seiyria, we've been trying to push this for years now but @isidorn doesn't live here anymore, he just comes past every now and then to close issues he doesn't want to work on. The fork on my profile allows sorting index files first, but I haven't kept the fork up to date with all the other changes since I made it.

**Comment by SimonDever** [comment]
It would be good to at least know why none of the options presented have been taken up. Or why they haven't at least been given creative criticism enough to improve. Also, is there a set of of acceptance criteria that we are unaware of? There is a lot of capability here.

**Comment by dzintars** [comment]
It would be nice to have option to sort by file comment.
```bash
setfattr -n user.comment -v "my-custom-tag-or-name" directory-or-filename
```
Read the attributes
```bash
getfattr directory-or-filename
```
And then in Settings could set something like
```json
{
"explorer.sortOrder": ["userComment", "default"],
}
```
which could enforce the sorting by user comment first if comment exists and all the other files sort at the bottom by default order.
I also see how regex could be incorporated there. (we don't talk about the performance there, right?) 😃 

P.S. This just like a dummy idea for the further research.

**Comment by SimonDever** [comment]
@isidorn, please review and consider some of this stuff instead of just closing duplicates. You haven't said one word in 4 years on why none of the PRs or suggestions are acceptable. Perhaps someone else at Microsoft should take over these particular issues if you don't have the time or inclination.

**Comment by andyp1per** [comment]
Apart from the tragedy that is cpptools performance this is the single biggest VSCode outage for me. I end up having lots of tabs open and it is impossible to find anything as they seem to be randomly ordered.

**Comment by bgracie** [comment]
My flow is to hide the tabs and just use open editors to manage multiple files.  This way, more room is available for all the open editors' paths and the top of the editor is cleanly reserved for the current editor file name and path.

I appreciate that we got alphabetical open editors sorting, but if it doesn't match the tabs sorting then it's not possible to easily navigate forward and back through the open editors using keyboard shortcuts, so I'm still stuck with the default ordering.

**Comment by Papooch** [comment]
What is the status on "Feature Request: api to custom files sort order at the sidebar"? I've been wanting this feature for years now and there is a considerable demand for it that I might even try to implement it myself if someone could introduce me to the project. @bpasero do you think it would be possible?

**Comment by drpicox** [comment]
> Working with TypeScript, it would be great to have the barrels (index.ts) sorted at the top.
> Would that be possible?

Agree: #128834

**Comment by MarcGuiselin** [comment]
I've made a PR that adds an `explorer.sortOrder` option `indexFirst` (default file sorting, but sorts files matching `**/index.*` first.)

**Comment by jeff-hykin** [comment]
>  developer complaints can be addressed by adding a few custom sorting options that addresses their issues.

Just for the record, adding another hardcoded option would not address my complaint

**Comment by MarcGuiselin** [comment]
> Just for the record, adding another hardcoded option would not address my complaint

Absolutely. I submitted that PR with the intent to bring attention to this issue, so that a proper solution be implemented, such as yours. I know the employees working on this repo are busy, but I'm just trying to get this issue on someone's radar...

You've handed MS a simple, short, backwards-compatible, community-approved solution on a silver platter and it was lost in a backlog. It's a bit disappointing.

**Comment by mrmokwa** [comment]
> I've made a PR that adds an `explorer.sortOrder` option `indexFirst` (default file sorting, but sorts files matching `**/index.*` first.)

Does somebody know if there is community extension to achieve this?

**Comment by SimonDever** [comment]
Can't be done, those APIs aren't open to the extensions. There are a few PRs with possible solutions though.

**Comment by ardijanr** [comment]
Sorting with special files first or last should be done for other languages too, like rust. Same sorting concept but file is called mod.rs.


Feature suggestion:

1. 
   Description: Sort special files first or last. 
   Dropdown options: First, Last

2. 
    Description: Special files list, case insensitive. Filnames that match the entire word of any words in this list will be sorted in the order they are written in. They will then be placed according to the "sort special files first or last" setting. Separate each filename with ';'

    String input field: mod.rs;index.ts;

**Comment by aswinkumar863** [comment]
I primarily work in MVC kind of projects, where I have to group controller files next to each other, followed by view files and so on. So, I'm also in need of this feature and I finally created a VS Code extension to solve this. You can either sort automatically or manually using my extension.

Check out my extension at here: https://marketplace.visualstudio.com/items?itemName=aswinkumar863.sort-editors

**Comment by SimonDever** [comment]
While cool, @aswinkumar863, that's an extension for editor tab sorting, not sorting of files in the explorer tree, which cannot be done with an extension.

**Comment by QuarticCat** [comment]
I pretty much like these two sorting behaviors in CLion:

- Header files and implementation files of the same name will be put together. For example, `xxx.h` and `xxx.cpp` will be put adjacent no matter what order you choose.
- `mod.rs` will be put on top of the files in a folder.

I hope the API can implement both of them.

**Comment by gulshan** [comment]
Can LSP be allowed to handle the sorting/hiding/nesting of the files and folders in the explorer?

**Comment by ivxvm** [comment]
It's been 5 years 4 months 2 days since this issue was opened and there is still no solution.

**Comment by acherkashin** [comment]
I would like to show `README.md` as a first element in sidebar to make it easy to find for new developers. In this case, they will be able to quickly look at the directory description.

**Comment by starball5** [comment]
Related: #119207, #91843, #95239.

Related questions on Stack Overflow:

- [Is there a way to organize files manually in VS Code's Explorer view?](https://stackoverflow.com/q/76009942/11107541)
- [How can I manually change the order that folders are listed in Visual Studio Code Explorer?](https://stackoverflow.com/q/54968639/11107541)

Related on r/vscode: [Is there a way to sort files manually?](https://www.reddit.com/r/vscode/comments/fcuygt/is_there_a_way_to_sort_files_manually/)

**Comment by malininss** [comment]
I'm still waiting for the ability for sorting index files. It is clear that the PR shouldn't be specified only to javascript, but this modification does not look like something complicated. It's possible to add a custom setting that accepts an array of files to be placed at the top of the tree.

If I add this feature, will the PR be accepted?

**Comment by JiveDig** [comment]
This issue seems to mention a few different sorting needs. I'm trying to sort all of the top level folders/directories/repos in my Explorer > Workspace in the sidebar. Is this currently possible with or without an extension? I can't figure out a way to do it.

**Comment by HonzaKubita** [comment]
I think that being capable of specifying the order of files in the text editor is very much a needed feature. I am currently working on a nuxt3 + firebase project and the root directory is getting flooded with folders that are in just random order without any respect what's vue what's css what's logic. I really like the way #119207 solves this issue.

**Comment by kilianc** [comment]
I'd pay for this one https://github.com/microsoft/vscode/issues/40679 :) 

```
a.go
a_test.go
a_linux.go
a_linux_test.go
```
instead of
```
a_linux_test.go
a_linux.go
a_test.go
a.go
```

**Comment by AmbrosiaDevelopments** [comment]
You'll have to pay to buy all of Microsoft to do anything about this. It's easy to keep a subscription to this thread, but I don't think it's ever going to change, we've been waiting 7 years already and they've had multiple PRs and offers of help. Never any constructive feedback. Oh and no extension can perform this behaviour because those APIs are not open to extensions.

**Comment by AmbrosiaDevelopments** [comment]
They've actually closed so many duplicates that as far as I can see there is no open issue left that is requesting the custom sort order people are asking for. This issue just lists all related issues and doesn't mention what its own intention is. Can someone find my an open issue that is still asking for custom sorting options or extension API? They keep closing issues saying they don't accept PRs without discussions first but I've been talking about it for 5 years. I have a PR and so do others and they keep saying we need to have community discussions. #141225 and #73930

**Comment by kilianc** [comment]
@AmbrosiaDevelopments is there a way to hack this into the editor so that people can unblock themselves?

**Comment by AmbrosiaDevelopments** [comment]
That's really interesting, I've never seen that before so I'm not sure just yet. It might take me a couple of weeks to get around to it but I'll give it a shot.

**Comment by gulshan** [comment]
As I have said, I think LSPs or other extensions should also be able to provide custom sorting, hiding and nesting logic for the file explorer. Then extensions will be able to provide some custom logical "view"s beside the default view in the file explorer. Extensions can already handle the nesting. It will remove the need of some custom file explorer widgets. For example, F# projects have their files included (and thus compiled) in a specific order. Now a separate widget is needed to show files in that order. If it was available in the file explorer, the separate widget would not be needed.

**Comment by gmanavarro** [comment]
It's been almost 7 years since this issue was opened and there is still no solution.

**Comment by AmbrosiaDevelopments** [comment]
Hi @lramos15, as per comments from @isidorn on #73930, could we please have a conversation about the change proposed by the PR? I have followed the patterns of the existing options and added tests and can see no adverse impacts. This issue we're now talking in doesn't have a purpose at the moment so can we bring the discussion around to this PR and what would be required to get something like this approved and merged? No other issue listed in the first comment is still open except to align behaviour to OS automatically. Do we need to open a new issue or a new PR covering more options than just index files? Would a pattern matching solution be more appropriate for broader coverage of everyone's issues?

**Comment by lramos15** [comment]
@AmbrosiaDevelopments We understand this is a highly requested ask.

What I'm trying to figure out is how to do this is an extensible way that covers everyone's use case as there are a lot of different desired sorting orders. There's two problems with extension API. The first is the extension host can be slow, leading to us being unable to render the tree at all until we get a sort order from the extension. The second one is how multiple extensions contributing a sort order interact.

**Comment by jeff-hykin** [comment]
> The second one is how multiple extensions contributing a sort order interact.

Have each extension register a sort-order names based on their extension ID. Ex: `username.extname:alphabetical`.

Then, in settings, let users pick the order of applicaiton.

settings.json
```jsonc
{
    "explorer.sortBy”: [
        "folders first",
        "username.extname:alphabetical",
    ]
}
```

**Comment by php4fan** [comment]
Issue 199676 should be reopened.

I'm commenting here because not only was it closed, but also locked to comments,  for no good reason.

Note that the default, out-of-the-box sorting of files, without tinkering with settings, should be a reasonable one, and it currently isn't.

**Comment by kai-353** [comment]
I wanted to make an extension that does something like this, but I don't think it is possible with the current VS Code API's. The closest extension I could find was [this one](https://marketplace.visualstudio.com/items?itemName=qingshun.explorer-sort-order) by Qing Shun, but it just switches between 2 sort options in the VS Code settings.

Commenting here to keep myself updated.

**Comment by gjsjohnmurray** [comment]
> I'd pay for this one [#40679](https://github.com/microsoft/vscode/issues/40679) :)

@kilianc can you use [Explorer file nesting](https://code.visualstudio.com/updates/v1_67#_explorer-file-nesting) for this? It first appeared in April 2022.

**Comment by canklot** [comment]
### SortMyFiles Extension 

After many years I wrote an extension for it. All you had to do it install the extension and write the file names you want to re-order in the `.order` file. Normally VS Code API doesn't support this. But I found kind of a work around. My extension plays with the last modified attributes of the files and changes the default sort order to last modified.  

https://github.com/canklot/SortMyFiles
https://marketplace.visualstudio.com/items?itemName=CanklotSoftware.SortMyFiles

**Comment by canklot** [comment]
> > ### SortMyFiles Extension
> > After many years I wrote an extension for it. All you had to do it install the extension and write the file names you want to re-order in the `.order` file. Normally VS Code API doesn't support this. But I found kind of a work around. My extension plays with the last modified attributes of the files and changes the default sort order to last modified.
> > https://github.com/canklot/SortMyFiles https://marketplace.visualstudio.com/items?itemName=CanklotSoftware.SortMyFiles
> 
> This is a great extension as a temporary fix for the issue but a proper solution in VSCode would be ideal so we don't need to use this last modified date workaround. It looks like this will prevent the rest of the explorer from being sorted alphabetically, correct?

 :blush: Not anymore. It took me a few days but I updated the code. Now it sorts rest of the files alphabetically

**Comment by lanly-dev** [comment]
I'm here to vote for **Feature Request: api to custom files sort order at the sidebar**

**Comment by CN120** [comment]
I found this [newly created extension](https://marketplace.visualstudio.com/items?itemName=RARI.custom-file-order) that allows for manual sorting of files in a new explorer pane. It looks really great! Would love to see the way this extension is done be built in natively because, no shade to the author but, I am not sure how my IT org will feel about me using an extension written by a high schooler from Mongolia.

**Comment by ZJunSong** [comment]
> I'm here to vote for **Feature Request: api to custom files sort order at the sidebar**

vote+1! I really need this: Custom my Folders and files!
